src/main/java/com/gooddata/Authentication.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2004-2016, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+/**
+ * HttpClient factory
```

Comment:
http client factory is an implementation detail, please explain purpose of this class